util.go

```diff
@@ -30,19 +31,24 @@ func makeEnclosingDirectories(path string) error {
 	return nil
 }
 
-func writeFile(reader io.Reader, path string) error {
+func writeFile(reader io.Reader, path string) (err error) {
 	if err := makeEnclosingDirectories(filepath.Dir(path)); err != nil {
 		return errors.Wrap(err, "problem making enclosing directories")
 	}
 
 	file, err := os.Create(path)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "problem creating file")
 	}
 
-	if _, err := io.Copy(file, reader); err != nil {
-		return err
-	}
+	catcher := grip.NewBasicCatcher()
+	defer func() {
```

Comment:
Unless there's a strong reason to put this logic in a defer, I think it would be better to simply have it in the end of the function.

Reply:
I moved the closes to all the return cases.